### ISSUE TITLE
Shim 15.6 for Cisco PuzzleOS

### ISSUE THREAD

**Issue by joylatten** [rhboot/shim-review#262]
Confirm the following are included in your repo, checking each box:

 - [X] completed README.md file with the necessary information
 - [X] shim.efi to be signed
 - [X] public portion of your certificate(s) embedded in shim (the file passed to VENDOR_CERT_FILE)
 - [X] binaries, for which hashes are added to vendor_db ( if you use vendor_db and have hashes allow-listed )
 - [X] any extra patches to shim via your own git tree or as files
 - [X] any extra patches to grub via your own git tree or as files
 - [X] build logs
 - [X] a Dockerfile to reproduce the build of the provided shim EFI binaries

-------------------------------------------------------------------------------
### What is the link to your tag in a repo cloned from rhboot/shim-review?
-------------------------------------------------------------------------------
`https://github.com/puzzleos/shim-review/tree/puzzleos-shim-x86_64-20220706`

-------------------------------------------------------------------------------
### What is the SHA256 hash of your final SHIM binary?
-------------------------------------------------------------------------------
c44d3bff9c43a24b443a8ba48cf8963441291b48e44c6e427d628e8a05a64e46  shimx64.efi

**Comment by steve-mcintyre** [comment]
Following on from #182 and #212. presumably

**Comment by steve-mcintyre** [comment]
Looking:
- ID already done in #212
- shim reproduces here
- shim from upstream, no patches
- SBAT data looks ok for shim
- no issues with revocation, new certs
- Complicated set of 3 embedded CA certs, each with **huge** lifetime (2071!) Thanks for at least showing how you generate the .esl file here!
- HSM for key management
- I remember looking at stubby before, looks ok
- linux setup sounds ok
- no grub

Queries:
- You appear to be embedding multiple CA keys. I think I understand why you have multiple keys, but why is the CA bit set?

**Comment by joylatten** [comment]
Hi @steve-mcintyre,
That CA bit was set because of how certs were generated. The 3 certs will not be used as CAs.

**Comment by steve-mcintyre** [comment]
Sorry, but that's a little disconcerting here for me. I'm always a bit worried to see things open wider than they need to be in this context.

@frozencemetery @julian-klode @vathpela what do you think here please?

**Comment by joylatten** [comment]
Sorry, meant to say they were generated as 3 self-signed CA certificates.

**Comment by frozencemetery** [comment]
At the very least I'd want it documented in the submission, which it currently isn't.

> That CA bit was set because of how certs were generated.

I mean, sure, but why though?  Why not generate them without that?

> The 3 certs will not be used as CAs.

I would think this presents an auditability problem: sure, you don't want to, but they can be - nothing stops them from signing another cert.

> Sorry, but that's a little disconcerting here for me. I'm always a bit worried to see things open wider than they need to be in this context.

I think I share this concern.

**Comment by joylatten** [comment]
We used self signed CA certificates in the proposed shim simply out of convention and consistency. Most of the accepted shim requests we looked at had CA certificates.
We would like to continue the review with CA in place.  If doing so is not an option, then we can patch our shim to embed self signed certificates without CA bit set.

**Comment by joylatten** [comment]
@steve-mcintyre, @frozencemetery ,
I have patched our shim to include our 3 self-signed certificates with the CA bit set to FALSE. This resulted in a new vendor_db.esl as well as a new shim hash which I have edited in the above checklist as well as in the ISSUE_TEMPLATE.md and in the README.md. I also included a sentence in the README.md referencing they are 3 self-signed certs with no CA.  
Otherwise, nothing else changed.
Thanks for the review and feedback.

**Comment by joylatten** [comment]
Hi. Any new status?

**Comment by steve-mcintyre** [comment]
Sorry to keep you waiting...
- verified new shim build checksum
- certs look good now, thanks

All done!

**Comment by joylatten** [comment]
Thanks!

**Comment by frozencemetery** [comment]
@joylatten Did you receive a signed shim?

**Comment by frozencemetery** [comment]
Well, closing in any case due to https://github.com/rhboot/shim-review/issues/307

**Comment by joylatten** [comment]
@frozencemetery , unfortunately,  we are waiting on Microsoft to sign our shim. 
Is #307 going to be an issue for us in 15.6?

**Comment by joylatten** [comment]
oh gosh, do I need to resubmit and use 15.7 with NX?

**Comment by steve-mcintyre** [comment]
If you have to re-submit then yes, you'll need to do NX with 15.7. What state is your submission in on  the MS partner site?

**Comment by joylatten** [comment]
its in the review state

**Comment by steve-mcintyre** [comment]
I think you may need to prod folks at MS about this then - normally things in "review" state get signed within a few days of the "accepted" label here,

**Comment by joylatten** [comment]
@steve-mcintyre , thanks, I will give it a try and cross my fingers. It has been in review for a few weeks now.
do you think they may accept it with it being 15.6? I am assuming 15.6 did not have NX enabled...?

**Comment by steve-mcintyre** [comment]
They have been happy to continue to sign things without nx, **so long as** they were submitted before the deadline.

**Comment by joylatten** [comment]
@steve-mcintyre ok, thanks. Hopefully they will consider my first submission is the same as the 2nd one, just the EV 
cert was incorrect.
